c7n/resources/rds.py

```diff
@@ -223,5 +224,66 @@ def process(self, resources):
                 raise
 
             self.log.info("Deleted rds: %s" % rdb['DBInstanceIdentifier'])
-        
-        
+
+
+@actions.register('snapshot')
+class Snapshot(BaseAction):
+
+    schema = {'properties': {
+        'type': {
+            'enum': ['snapshot']}}}
+
+    def process(self, resources):
+        with self.executor_factory(max_workers=3) as w:
+            futures = []
+            for resource in resources:
+                futures.append(w.submit(
+                    self.process_rds_snapshot,
+                    resource))
+                for f in as_completed(futures):
+                    if f.exception():
+                        self.log.error(
+                            "Exception creating rds snapshot  \n %s" % (
+                                f.exception()))
+
+    def process_rds_snapshot(self, resource):
+        c = local_session(self.manager.session_factory).client('rds')
+        c.create_db_snapshot(
+            DBSnapshotIdentifier="BKUP-%s-%s" % (
+                resource['DBInstanceIdentifier'],
+                resource['Engine']),
+            DBInstanceIdentifier=resource['DBInstanceIdentifier'])
+
+
+@actions.register('retention')
+class RetentionWindow(BaseAction):
+
+    date_attribute = "BackupRetentionPeriod"
+    schema = type_schema('retention', days={'type': 'number'})
+
+    def process(self, resources):
+        with self.executor_factory(max_workers=3) as w:
+            futures = []
+            for resource in resources:
+                futures.append(w.submit(
+                    self.process_rds_snapshot,
+                    resource))
+                for f in as_completed(futures):
+                    if f.exception():
+                        self.log.error(
+                            "Exception setting rds retention  \n %s" % (
+                                f.exception()))
+
+    def process_snapshot_retention(self, resource):
+        try:
+            v = int(resource['BackupRetentionPeriod'])
```

Comment:
instead of try except here you can do v = int(resource.get('BackupRetentionPeriod', 0))